src/server.h

```diff
@@ -2189,6 +2189,7 @@ struct valkeyServer {
     int cluster_slot_stats_enabled;                        /* Cluster slot usage statistics tracking enabled. */
     /* Debug config that goes along with cluster_drop_packet_filter. When set, the link is closed on packet drop. */
     uint32_t debug_cluster_close_link_on_packet_drop : 1;
+    uint32_t debug_cluster_random_min_ping : 1;
```

Comment:
What does this mean of min-ping?

Reply:
it is for this random ping logic, i can not think of a better word.
```
    /* Ping some random node 1 time every 10 iterations, so that we usually ping
     * one random node every second. */
    if (server.debug_cluster_random_min_ping && !(iteration % 10)) {
        int j;

        /* Check a few random nodes and ping the one with the oldest
         * pong_received time. */
        for (j = 0; j < 5; j++) {
            de = dictGetRandomKey(server.cluster->nodes);
            clusterNode *this = dictGetVal(de);

            /* Don't ping nodes disconnected or with a ping currently active. */
            if (this->link == NULL || this->ping_sent != 0) continue;
            if (this->flags & (CLUSTER_NODE_MYSELF | CLUSTER_NODE_HANDSHAKE)) continue;
            if (min_pong_node == NULL || min_pong > this->pong_received) {
                min_pong_node = this;
                min_pong = this->pong_received;
            }
        }
        if (min_pong_node) {
            serverLog(LL_DEBUG, "Pinging node %.40s", min_pong_node->name);
            clusterSendPing(min_pong_node->link, CLUSTERMSG_TYPE_PING);
        }
    }
```

Reply:
So it means *ping one random node every second*?

Reply:
yes.

Reply:
So why "min"? Is it a minimum?

> it is for this random ping logic, i can not think of a better word.

I think you said a better word yourself: "random ping". Or another variant: "ping random node".

```suggestion
    uint32_t debug_cluster_random_ping : 1;
```

Reply:
ok, good, i mentioned min earlier probably because this min_pong_node naming.

```
            if (min_pong_node == NULL || min_pong > this->pong_received) {
                min_pong_node = this;
                min_pong = this->pong_received;
            }
```

Reply:
Ah, min_pong_node means minimum pong_received timestamp, the node we didn't get a pong from for a long time. I understand now, but I think "min" is not important for the name of the debug flag.